### PR TITLE
support config file for cdc

### LINKED ISSUE #3995: Support passing raw toml config for ticdc as string type
## Feature Request

**Is your feature request related to a problem? Please describe:**

Now all the configurations for PD/TiDB/TiKV/TiFlash/Pump have supported raw toml format, refer to #3342, #3327, #3312, we also need to support this for TiCDC.
**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by july2993**
### What problem does this PR solve?

fix #3995 
### What is changed and how does it work?

Make `config` field of cdc spec support both `string` and `object` type, keep behavior like before if it's an object.
otherwise, generate the config file and use it.(will still use the flags we used before `--gc-ttl` `--log-info` `--log-level`)
### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [x] E2E test 
- [x] Manual test 
### do not specified `config`
```yaml
    ticdc:
      baseImage: pingcap/ticdc
      replicas: 1
```
check cdc starts like before(no `--config` flag)
### specified config like before
```yaml
    ticdc:
      baseImage: pingcap/ticdc
      replicas: 1
      config:
        timezone: UTC
        gcTTL: 10000
        logLevel: debug
```
check cdc starts like before(no `--config` flag)

### specified raw config of cdc
```yaml
    ticdc:
      baseImage: pingcap/ticdc
      replicas: 1
      config: |
        capture-session-ttl = 1000
```
check generate configmap and use `--config` flag
```
➜  tidb-operator git:(cdc_config) ✗ k exec -it test-ticdc-0 -- sh
/ # ps
PID   USER     TIME  COMMAND
    1 root      1:01 /cdc server --addr=0.0.0.0:8301 --advertise-addr=test-ticdc-0.test-ticdc-peer.default.svc:8301 --gc-ttl=86400 --log-file=/dev/stderr --log-level=info --pd=http://test-pd:2379 --config=/etc/ticdc/ticdc.toml
   34 root      0:00 sh
   41 root      0:00 ps
/ # cat /etc/ticdc/ticdc.toml
capture-session-ttl = 1000
```
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Support passing raw config file data to cdc
```

### CODE REVIEW

File: tests/e2e/tidbcluster/tidbcluster.go [hunk 1]
```diff
@@ -1481,6 +1481,15 @@ var _ = ginkgo.Describe("TiDBCluster", func() {
 		err = oa.WaitForTidbClusterReady(fromTc, 30*time.Minute, 5*time.Second)
 		framework.ExpectNoError(err, "Expected TiDB cluster ready")
 
+		ginkgo.By("Update cdc config to use config file")
```
Comment (DanielZhangQD) [inline tests/e2e/tidbcluster/tidbcluster.go:1484]: Suggest checking the config file here, refer to https://github.com/pingcap/tidb-operator/blob/master/tests/e2e/tidbcluster/tidbcluster.go#L1667-L1677.

Comment (july2993) [reply tests/e2e/tidbcluster/tidbcluster.go:1484]: https://github.com/pingcap/tidb-operator/pull/4010/commits/a61ed34931c2491cfbde3ef374dda0bc527ef3c0

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Review by handlerww — APPROVED** [review]
LGTM

**Pull request merged**